client/modules/core/main.js

```diff
@@ -483,10 +483,12 @@ export default {
   },
 
   setShopId(id) {
-    if (id) {
-      this.shopId = id;
-      this.setUserPreferences("reaction", "activeShopId", id);
-    }
+    if (!id) { return; }
+
+    this.shopId = id;
+    this.setUserPreferences("reaction", "activeShopId", id);
+
+    Meteor.call("shop/resetShopId");
```

Comment:
to maintain consistency between client and server, I added the `"shop/resetShopId"` method to effectively clear cache, and re-determine the shopId. I chose not to do a _set_ shopId method to avoid having to check whether the user is permitted to do so (and presumably duplicating logic which already exists).